src/fuselage/CCPACSDoorAssemblyPosition.cpp

```diff
@@ -16,17 +16,171 @@
+    builder.Add(cutCompound, sStringer.GetCutGeometry(FUSELAGE_COORDINATE_SYSTEM));
+    builder.Add(cutCompound, eStringer.GetCutGeometry(FUSELAGE_COORDINATE_SYSTEM));
+
+    // split fuselage loft
+    const TopoDS_Shape loft         = m_parent->GetParent()->GetParent()->GetLoft()->Shape();
+    const TopoDS_Shape splittedLoft = SplitShape(loft, cutCompound);
+
+    // find door face
+    TopTools_IndexedMapOfShape faceMap;
+    TopExp::MapShapes(splittedLoft, TopAbs_FACE, faceMap);
+
+    const CTiglTransformation& trafo = m_parent->GetParent()->GetParent()->GetTransformationMatrix();
+
+    TopoDS_Compound compound;
+    builder.MakeCompound(compound);
+    for (int i = 1; i <= faceMap.Extent(); i++) {
+        const TopoDS_Shape f = faceMap(i);
+        if (Contains(TopoDS::Face(trafo.Transform(f)))) {
+            builder.Add(compound, f);
+        }
+    }
+
+    m_geometry = compound;
+}
+
+void CCPACSDoorAssemblyPosition::UpdateBorders()
+{
+    CCPACSFrame& sFrame               = m_uidMgr->ResolveObject<CCPACSFrame>(m_startFrameUID);
+    CCPACSFrame& eFrame               = m_uidMgr->ResolveObject<CCPACSFrame>(m_endFrameUID);
+    CCPACSFuselageStringer& sStringer = m_uidMgr->ResolveObject<CCPACSFuselageStringer>(m_startStringerUID);
+    CCPACSFuselageStringer& eStringer = m_uidMgr->ResolveObject<CCPACSFuselageStringer>(m_endStringerUID);
+
+    m_borderCache.emplace();
+    BorderCache& c = *m_borderCache;
+
+    // get the intersection Points between stringer and frames
+    UpdateBorder(c.sFrame_sStringer, sFrame.GetGeometry(true), sStringer.GetGeometry(true));
+    UpdateBorder(c.sFrame_eStringer, sFrame.GetGeometry(true), eStringer.GetGeometry(true));
+    UpdateBorder(c.eFrame_sStringer, eFrame.GetGeometry(true), sStringer.GetGeometry(true));
+    UpdateBorder(c.eFrame_eStringer, eFrame.GetGeometry(true), eStringer.GetGeometry(true));
+
+    // generate directions to the opposite intersection point
+    c.sFrame_sStringer.SetDirection(gp_Vec(c.sFrame_sStringer.Location(), c.eFrame_eStringer.Location()));
+    c.sFrame_eStringer.SetDirection(gp_Vec(c.sFrame_eStringer.Location(), c.eFrame_sStringer.Location()));
+    c.eFrame_sStringer.SetDirection(gp_Vec(c.eFrame_sStringer.Location(), c.sFrame_eStringer.Location()));
+    c.eFrame_eStringer.SetDirection(gp_Vec(c.eFrame_eStringer.Location(), c.sFrame_sStringer.Location()));
+}
+
+void CCPACSDoorAssemblyPosition::UpdateBorder(gp_Ax1& b, TopoDS_Shape frame, TopoDS_Shape stringer)
+{
+    TopTools_IndexedMapOfShape edgeMap1;
+    edgeMap1.Clear();
+    TopExp::MapShapes(frame, TopAbs_EDGE, edgeMap1);
+
+    TopTools_IndexedMapOfShape edgeMap2;
+    edgeMap2.Clear();
+    TopExp::MapShapes(stringer, TopAbs_EDGE, edgeMap2);
+
+    for (int i = 1; i <= edgeMap1.Extent(); i++) {
+        for (int j = 1; j <= edgeMap2.Extent(); j++) {
+            BRepExtrema_ExtCC pG(TopoDS::Edge(edgeMap1(i)), TopoDS::Edge(edgeMap2(j)));
```

Comment:
What, if there are multiple intersections between the stringer and the frame? Can this happen?